.github/workflows/misc.yml

```diff
@@ -0,0 +1,120 @@
+name: XGBoost CI (misc)
+
+on: [push, pull_request]
+
+permissions:
+  contents: read  # to fetch code (actions/checkout)
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+env:
+  BRANCH_NAME: >-
+    ${{ github.event.pull_request.number && 'PR-' }}${{ github.event.pull_request.number || github.ref_name }}
+
+jobs:
+  gtest-cpu:
+    name: Test Google C++ test (CPU)
+    runs-on: macos-13
+    steps:
+      - uses: actions/checkout@v4.2.2
```

Comment:
Is it important to pin to micro version? Would v4 suffice? If so, it would give you more leeway in terms of getting updates and fixes on the action.

Reply:
I revised all pins to specify the macro version only.